domain/usecases/unsubscribe_test.go

```diff
@@ -0,0 +1,117 @@
+package usecases
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/matheusmosca/walrus/domain/entities"
+	"github.com/matheusmosca/walrus/domain/vos"
+)
+
+func TestUnsubscribe(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		ctx     context.Context
+		message vos.Message
+	}
+
+	type fields struct {
+		storage   Repository
+		topicName vos.TopicName
+	}
+
+	tests := []struct {
+		name           string
+		args           args
+		fields         fields
+		beforeRun      func(topic entities.Topic) (chan vos.Message, vos.SubscriberID)
+		wantErr        error
+		wantSubscriber bool
+	}{
+		{
+			name: "unsubscribe should succeed",
+			args: args{
+				ctx: context.Background(),
+				message: vos.Message{
+					TopicName: "unsubscribing",
+				},
+			},
+			fields: fields{
+				storage:   &RepositoryMock{},
+				topicName: "unsubscribing",
+			},
+			beforeRun: func(topic entities.Topic) (chan vos.Message, vos.SubscriberID) {
+				subscriber := entities.NewSubscriber(topic)
+				ch, ID := subscriber.Subscribe()
```

Comment:
could you write `id` in lowercase? I think variables should not start with capitalized letters.
```suggestion
				ch, id := subscriber.Subscribe()
```